mm2src/coins/lp_coins.rs

```diff
@@ -898,17 +900,46 @@ struct CoinsContext {
     /// Similar to `LP_coins`.
     coins: AsyncMutex<HashMap<String, MmCoinEnum>>,
     balance_update_handlers: AsyncMutex<Vec<Box<dyn BalanceTradeFeeUpdatedHandler + Send + Sync>>>,
+    tx_history_path: PathBuf,
+    /// The database has to be initialized only once!
+    /// It's better to use something like [`Constructible`], but it doesn't provide a method to get the inner value by the mutable reference.
+    tx_history_db: AsyncMutex<Option<TxHistoryDb>>,
 }
 impl CoinsContext {
     /// Obtains a reference to this crate context, creating it if necessary.
     fn from_ctx(ctx: &MmArc) -> Result<Arc<CoinsContext>, String> {
+        let tx_history_path = ctx.dbdir().join("TRANSACTIONS");
         Ok(try_s!(from_ctx(&ctx.coins_ctx, move || {
             Ok(CoinsContext {
                 coins: AsyncMutex::new(HashMap::new()),
                 balance_update_handlers: AsyncMutex::new(vec![]),
+                tx_history_path,
+                tx_history_db: AsyncMutex::new(None),
             })
         })))
     }
+
+    async fn tx_history_db(&self) -> TxHistoryResult<TxHistoryDbLocked<'_>> {
+        /// # Safe
+        ///
+        /// Make sure the inner value of the `guard` is `Some`, i.e. `guard.is_some()`.
+        unsafe fn unwrap_tx_history_db(guard: AsyncMutexGuard<'_, Option<TxHistoryDb>>) -> TxHistoryDbLocked<'_> {
```

Comment:
What is the reason to mark this fn as unsafe? As per https://doc.rust-lang.org/book/ch19-01-unsafe-rust.html#unsafe-superpowers unsafe typically means *memory* unsafe code, which is not the case here. It can panic, so according to https://rust-lang.github.io/api-guidelines/documentation.html#function-docs-include-error-panic-and-safety-considerations-c-failure, we should use `# Panics` annotation instead.

Reply:
It makes sense! I wanted to mark it as unsafe just to highlight the use of the function... You are right, it's better to add the `# Panics` annotation instead!

Reply:
Done